src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,354 @@
+    public volatile Map<RepairType, Options> repair_type_overrides = new EnumMap<>(RepairType.class);
+
+    public AutoRepairConfig()
+    {
+        this(false);
+    }
+
+    public AutoRepairConfig(boolean enabled)
+    {
+        this.enabled = enabled;
+        global_settings = Options.getDefaultOptions();
+        for (RepairType type : RepairType.values())
+        {
+            repair_type_overrides.put(type, new Options());
+        }
+    }
+
+    public int getRepairCheckIntervalInSec()
+    {
+        return repair_check_interval_in_sec;
+    }
+
+    public boolean isAutoRepairSchedulingEnabled()
+    {
+        return enabled;
+    }
+
+    public int getAutoRepairHistoryClearDeleteHostsBufferInSec()
+    {
+        return history_clear_delete_hosts_buffer_in_sec;
+    }
+
+    public void setAutoRepairHistoryClearDeleteHostsBufferInSec(int seconds)
+    {
+        history_clear_delete_hosts_buffer_in_sec = seconds;
+    }
+
+    public boolean isAutoRepairEnabled(RepairType repairType)
+    {
+        return enabled && applyOverrides(repairType, opt -> opt.enabled);
+    }
+
+    public void setAutoRepairEnabled(RepairType repairType, boolean enabled)
+    {
+        if (enabled && repairType == RepairType.incremental &&
+            (DatabaseDescriptor.getMaterializedViewsEnabled() || DatabaseDescriptor.isCDCEnabled()))
+            throw new ConfigurationException("Cannot enable incremental repair with materialized views or CDC enabled");
+
+        repair_type_overrides.get(repairType).enabled = enabled;
+    }
+
+    public void setRepairByKeyspace(RepairType repairType, boolean repairByKeyspace)
+    {
+        repair_type_overrides.get(repairType).repair_by_keyspace = repairByKeyspace;
+    }
+
+    public boolean getRepairByKeyspace(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.repair_by_keyspace);
+    }
+
+    public int getRepairThreads(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.number_of_repair_threads);
+    }
+
+    public void setRepairThreads(RepairType repairType, int repairThreads)
+    {
+        repair_type_overrides.get(repairType).number_of_repair_threads = repairThreads;
+    }
+
+    public int getRepairSubRangeNum(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.number_of_subranges);
+    }
+
+    public void setRepairSubRangeNum(RepairType repairType, int repairSubRanges)
+    {
+        repair_type_overrides.get(repairType).number_of_subranges = repairSubRanges;
+    }
+
+    public int getRepairMinIntervalInHours(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.min_repair_interval_in_hours);
+    }
+
+    public void setRepairMinIntervalInHours(RepairType repairType, int repairMinFrequencyInHours)
+    {
+        repair_type_overrides.get(repairType).min_repair_interval_in_hours = repairMinFrequencyInHours;
```

Comment:
JFYI, there's a bug in these setters. If you provide a config in `cassandra.yaml` such as the following

```
auto_repair:
  repair_type_overrides:
    full:
      enabled: true
```

`repair_type_overrides.get("incremental")`  will be null and we'll get an NPE if we call a setter with incremental repair type

Reply:
Though this only affects nodetool functionality